### PR TITLE
server: fix start ts in slow log when transaction retrying

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What problem does this PR solve? 
If the transaction retried, the start ts in slow log will be the start ts of the last successful retried transaction.

### What is changed and how it works?
Record the start ts before the transaction to be committed.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

1. start a TiDB server
1. `set tidb_disable_txn_auto_retry = 0;`
1. `set tidb_slow_log_threshold = 0;`
1. 
    ```sql
    create table t (i int);
    insert into t values (1);
    begin;
    update t set i = 2;
    -- update t set i = 3; in another connection
    commit; -- the transaction will retry
    ```
1. `select * from slow_query where txn_start_ts = (select txn_start_ts from slow_query where query = 'update t set i = 2;') order by time \G`

Code changes

 - Has exported function/method change

Side effects

 - Increased code complexity

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: session/tidb.go [hunk 1]
```diff
@@ -210,11 +190,17 @@ func runStmt(ctx context.Context, sctx sessionctx.Context, s sqlexec.Statement)
 		ctx = opentracing.ContextWithSpan(ctx, span1)
 	}
 	se := sctx.(*session)
+	sessVars := se.GetSessionVars()
+	var startTS uint64
```
Comment (lysu) [inline session/tidb.go]: how about define a `type tsCollector struct { Ts []uint64 }` and pass it by `ctx` to  collect all retry ts?

Comment (jackysp) [reply session/tidb.go]: maybe we could add a new field for retry ts in slow log

Comment (coocood) [inline session/tidb.go]: We can store the current `TxnCtx` and use it in the defer function.

**Comment by coocood** [comment]
LGTM

**Pull request merged**